3scale/porta
# Load widgets in Dashboard only when opening a Service

**What this PR does / why we need it**:

Services charts are loaded even when they are not visible. The higher the number of services, the more time the UI takes to load.

This is just a first step to improve Dashboard's performance.

**Which issue(s) this PR fixes** 

[THREESCALE-2916: Reduce dashboard loading time](https://issues.jboss.org/browse/THREESCALE-2916)

**Verification steps** 

* Services that are not open (charts not visible) should not send any XHR (see browser's network tab)
* When opening a service, requests should be sent and charts should load eventually.
* When a service is open by default (at loading time), the charts should also be loaded automatically.

**Performance results**
> With 28 services.
* Before (all widgets are loaded):
<img width="1438" alt="Screen Shot 2019-07-10 at 17 58 04" src="https://user-images.githubusercontent.com/11672286/60984755-85d1df80-a33c-11e9-9718-034f373fe267.png">

* After (only visible widgets load):
<img width="1438" alt="Screen Shot 2019-07-10 at 17 59 27" src="https://user-images.githubusercontent.com/11672286/60984726-7c487780-a33c-11e9-99ce-10428fcddbd6.png">

The background work is dramatically reduced, although it is all non-blocked time, keep in mind this had been recorded with only 28 services.
